modin/engines/base/frame/data.py

```diff
@@ -1686,7 +1691,9 @@ def broadcast_apply_full_axis(
             None,
             None,
             dtypes,
-            validate_axes="all" if new_partitions.size != 0 else False,
+            validate_axes="all"
+            if any(o is not None for o in [new_index, new_columns])
```

Comment:
Well, previously we were validated axes almost in all cases of calls `broadcast_apply_full_axis`, because `new_partitions.size` usually (almost always) not zero. So, if we precomputed axes above (when `new_index` and `new_columns` was `None`) we don't need to validate axes. We don't need to validate them even when `new_index` and `new_columns` is not `None`, because it's our responsibility to pass new valid axes. However, when I've removed here validation at all, many tests started to fail in different parts of Modin, so I think completely remove validation is a separate issue.

I've done these changes here is to speed up common cases, when we're not passing `new_index` and `new_columns` (it's also a groupby case). So other cases should not degrade in speed.

Reply:
I see, it makes sense on removing validation for some cases in separate issue.